### DISCUSSION TITLE
HPN-SSH 18.3.1 Released.

### DISCUSSION THREAD

**Discussion by rapier1** [rapier1/openssh-portable#68] [Announcements]
This is a bugfix release. We discovered that under some versions of GCC13 (specifically in recent updates to Fedora based distros) the 128bit XOR we had been using in the AES-CTR parallel cipher was causing a segfault. We don't know if this is a bug in GCC or if they are enforcing new parameters. This is in a critical code path so we are providing a temporary (hopefully) fix where we disable the use of the 128bit XOR method. 

When we have a better handle on what is happen we hope to have a more effective fix. When that happens we will make a new release but we wanted to get this out as soon as possible. This should not significantly impact performance in most use cases.

**Comment by AEI-SteffenG** [comment]
I may be missing something really important, but where did the hpn-x.y.patch files go? Do I have to build it on my own, as a diff against the original tarball?

**Reply by dave-re-imprivata** [reply]
They were migrated to tags in the repo a while back. The easiest way to get a patch file is to make sure you have the latest tags, etc, and then do something like:

`git diff V_9_6_P1..hpn-18.3.1 > hpn-18.3.1.patch`

This diffs the official 9.6p1 release vs. the hpn-18.3.1 tag (so, full 18.3.1 features for 9.6p1) - adjust OpenSSH release tag and HPN release tag to suit your needs...

**Reply by AEI-SteffenG** [reply]
Perfect! Thanks

**Reply by rapier1** [reply]
So the patchfiles traditionally lived on sourgeforge at https://sourceforge.net/projects/hpnssh/files/Patches/
However, I've not been very good at keeping that repository up to date and I am sorry about that. I'll try to get some patch set up there shortly. That said, Dave is right, the best way to generate a patch set is to use `git diff` in order to ensure you have the most up to date patch set possible. 

I will note that knowing how to generate the patches has become more opaque than it used to be when we changed our versioning method. I need to think about the best way to indicate which hpn versions corresponds to which OpenSSH version. In the meantime the best way to find that out is to take a look at version.h in the master branch. This will *always* have the OpenSSH version included. E.g. 
```
/* $OpenBSD: version.h,v 1.100 2023/12/18 14:48:44 djm Exp $ */

#define SSH_VERSION	"OpenSSH_9.6"

#define SSH_PORTABLE	"p1"
#define SSH_HPN         "-hpn18.3.1"
#define SSH_RELEASE	SSH_VERSION SSH_PORTABLE SSH_HPN
```
